x-pack/plugins/security/server/config.ts

```diff
@@ -5,11 +5,24 @@
  */
 
 import crypto from 'crypto';
+import type { Duration } from 'moment';
 import { schema, Type, TypeOf } from '@kbn/config-schema';
 import { i18n } from '@kbn/i18n';
 import { Logger, config as coreConfig } from '../../../../src/core/server';
+import type { AuthenticationProvider } from '../common/types';
 
 export type ConfigType = ReturnType<typeof createConfig>;
+type RawConfigType = TypeOf<typeof ConfigSchema>;
+
+interface ProvidersCommonConfigType {
```

Comment:
__note:__ I realized that with previous types we are losing the actual types and end up with `any` in many places.